nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -401,6 +418,27 @@ void DockingServer::doInitialPerception(Dock * dock, geometry_msgs::msg::PoseSta
   }
 }
 
+
+
+void DockingServer::rotateToDock(const geometry_msgs::msg::PoseStamped & dock_pose){
+  rclcpp::Rate loop_rate(controller_frequency_);
+  geometry_msgs::msg::PoseStamped robot_pose = getRobotPoseInFrame(dock_pose.header.frame_id);
+  double angle_to_goal;
+  auto command = std::make_unique<geometry_msgs::msg::TwistStamped>();
+  while(rclcpp::ok()){
+    angle_to_goal = angles::shortest_angular_distance(tf2::getYaw(robot_pose.pose.orientation),
+    atan2(robot_pose.pose.position.y - dock_pose.pose.position.y,
+    robot_pose.pose.position.x - dock_pose.pose.position.x));
+    if(fabs(angle_to_goal) > 0.1){
```

Comment:
This should either be (1) a configurable threshold or (2) wait until its exceeded so its at least that much of a turn

Reply:
I made a mistake here, there should be an opposite sign (<)

Reply:
I called it tolerance in a hurry, but tell me please if it better should be "threshold"